testing/fetcher.ts

```diff
@@ -230,6 +231,30 @@ export class AuthenticatingFetcher implements Fetcher {
           headers: {...headers, Authorization: `Basic ${encodedAuth}`},
         };
       }
+      case AuthenticationType.Custom: {
+        const request = {
+          url,
+          body,
+          form,
+          headers,
+        };
+        const {params} = this._credentials as CustomCredentials;
+
+        let requestWithTemplateSubstitutions = JSON.stringify(request);
+        if (requestWithTemplateSubstitutions) {
+          // For awful APIs that require auth parameters in the request body, we have
+          // this scheme where we do template substitution of the body using an unguessable
+          // random token as part of the template key.
+          Object.entries(params).forEach(([key, value]) => {
```

Comment:
I wonder if we care to look at the params from the auth definition or not. For the most part, the set of params from the auth def should be exactly the same as the keys in the secret credentials, since the latter is derived from the former. But if you're changing your auth definition, they will diverge until you update the secrets. I suppose during that transition we could continue to use the old secrets verbatim, vs ignoring any keys that aren't present in the current auth definition. I think either way your requests probably won't work until the secrets are updated. So possibly fine as-is.

Reply:
Yeah this is how we currently do it for `MultiQueryParam` too where we store the actual "expected keys" from the current definition of the pack as well as the "stored key-value pairs" to handle when these values change. 

I'm not sure if we have any flow of invalidating/reauthenticating a user/system account connection tho when thes values change.. perhaps something for us to think about as a smooth item later on cc @alicia-codaio @punit-codaio